griptape/drivers/vector/pgvector_vector_store_driver.py

```diff
@@ -0,0 +1,159 @@
+import uuid
+from typing import Optional
+from attr import define, field, Factory
+from griptape.drivers import BaseVectorStoreDriver
+from sqlalchemy.engine import Engine
+from sqlalchemy import create_engine, Column, String, JSON
+from sqlalchemy.ext.declarative import declarative_base
+from sqlalchemy.dialects.postgresql import UUID
+from sqlalchemy.orm import Session
+from pgvector.sqlalchemy import Vector
+
+
+@define
+class PgVectorVectorStoreDriver(BaseVectorStoreDriver):
+    """
+    A vector store driver to Postgres using the PGVector extension.
+
+    Attributes:
+        connection_string: The connection string for the Postgres database instance.
+        create_engine_params: Additional configuration params passed when creating the database connection.
+        engine: An optional sqlalchemy Postgres engine to use.
+    """
+
+    connection_string: Optional[str] = field(default=None, kw_only=True)
+    create_engine_params: Optional[dict] = field(factory=dict, kw_only=True)
+    engine: Optional[Engine] = field(
+        default=Factory(
+            lambda self: create_engine(self.connection_string, **self.create_engine_params), takes_self=True
+        )
+    )
+
+    class VectorModel(declarative_base()):
```

Comment:
I _think_ add `@dataclass` decorator. Not sure `declarative_base()` changes anything.

Reply:
I believe `declarative_base()` provides the `.metadata.create_all()` method that we use to create the schema.